### PR TITLE
e2e test fixes to make them work on AWS

### PR DESCRIPTION
**Pull request by oavdeev**
**What this PR does / why we need it**:
Add CI/CD pipeline for AWS.

Notable changes: 
* shade kafka in the ingestion jar
* add boto3 to the feast-ci image
* tweak some timeouts in the e2e test suite

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: infra/scripts/test-end-to-end-aws.sh [hunk 1]
```diff
@@ -1,3 +1,19 @@
 #!/usr/bin/env bash
 
-aws sts get-caller-identity
\ No newline at end of file
+set -euo pipefail
```
Comment (woop) [inline infra/scripts/test-end-to-end-aws.sh:3]: Where is this file being used?

Comment (oavdeev) [reply infra/scripts/test-end-to-end-aws.sh:3]: This runs in a k8s cluster in AWS account, kicked off by codebuild job (the configuration for the codebuild job itself is not part of this PR).

**Review by woop — APPROVED** [review]

**Pull request merged**